addons/mail/wizard/mail_compose_message.py

```diff
@@ -142,12 +141,170 @@ def default_get(self, fields):
+        With template
+          * mass mail: take IDs from template
+        """
+        for composer in self:
+            res_ids = composer._parse_res_ids(composer.res_ids)
+            if composer.composition_mode == 'mass_mail' and composer.template_id and composer.template_id.attachment_ids:
+                composer.attachment_ids = composer.template_id.attachment_ids
+            elif composer.composition_mode == 'comment' and len(res_ids) == 1 and composer.template_id:
+                rendered_values = composer._generate_template_for_composer(
+                    res_ids,
+                    ('attachment_ids', 'attachments'),
+                )[res_ids[0]]
+                # transform attachments into attachment_ids; not attached to the
+                # document because this will be done further in the posting
+                # process, allowing to clean database if email not send
+                attachment_ids = []
+                if rendered_values.get('attachments'):
+                    attachment_ids = self.env['ir.attachment'].create([
+                        {'name': attach_fname,
+                         'datas': attach_datas,
+                         'res_model': 'mail.compose.message',
+                         'res_id': 0,
+                         'type': 'binary',    # override default_type from context, possibly meant for another model!
+                        } for attach_fname, attach_datas in rendered_values.pop('attachments')
+                    ]).ids
+                if rendered_values.get('attachment_ids'):
+                    attachment_ids += rendered_values['attachment_ids']
+                if attachment_ids:
+                    composer.attachment_ids = attachment_ids
+            else:
+                composer.attachment_ids = False
+
+    @api.depends('template_id', 'author_id', 'composition_mode', 'model', 'res_ids')
+    def _compute_email_from(self):
+        for composer in self:
+            if composer.template_id.email_from:
+                composer._set_value_from_template('email_from')
+            elif composer.author_id and not composer.email_from:
+                composer.email_from = composer.author_id.email_formatted
+
+    @api.depends('email_from', 'composition_mode', 'model', 'res_ids')
+    def _compute_author_id(self):
+        for composer in self:
+            if not composer.author_id:
+                if composer.email_from and composer.composition_mode == 'comment' and not composer.template_id:
+                    author = self.env['mail.thread']._mail_find_partner_from_emails([composer.email_from])[0]
+                else:
+                    author = self.env.user.partner_id
+                composer.author_id = author.id
+                # ensure email_from is set (inter dependent fields are hard to model
+                # in a compute)
+                if not composer.email_from:
+                    composer.email_from = author.email_formatted
```

Comment:
Would something like this work?

```suggestion
    @api.depends('template_id', 'author_id', 'email_from', 'composition_mode', 'model', 'res_ids')
    def _compute_sender_fields(self):
        for composer in self:
            if composer.template_id.email_from:
                composer._set_value_from_template('email_from')
            elif composer.author_id and not composer.email_from:
                composer.email_from = composer.author_id.email_formatted
            elif not composer.author_id:
                if composer.email_from and composer.composition_mode == 'comment' \
                        and not composer.template_id:
                    author = self.env['mail.thread']._mail_find_partner_from_emails([composer.email_from])[0]
                else:
                    author = self.env.user.partner_id
                composer.author_id = author.id
```